absl/base/BUILD.bazel

```diff
@@ -17,6 +17,7 @@
 load(
     "//absl:copts.bzl",
     "ABSL_DEFAULT_COPTS",
+    "ABSL_DEFAULT_LINKOPTS",
```

Comment:
I discussed this further with the team.  We aren't sure this viral ABSL_DEFAULT_LINKOPTS thing is the correct approach.  In fact, we've even started to reconsider whether ABSL_DEFAULT_COPTS is the correct approach.

Another possible approach is to use a CROSSTOOL file for this.  See https://github.com/bazelbuild/bazel/wiki/About-the-CROSSTOOL and https://github.com/bazelbuild/bazel/blob/master/tools/cpp/CROSSTOOL#L114.  There is even a way to set a custom CROSSTOOL file as the default for a Bazel WORKSPACE.

Until we resolve this, can you change this pull request to only include the changes to C++ files?  By this is mean limit this pull request to `defined(__FreeBSD__)` and the like.

You then can work around the linker issues with `bazel build --linkopt=-lm --linkopt=-pthread //path/to:target` or by defining a custom CROSSTOOL file for your project.

We can then open a separate issue where we can figure out the correct thing to do with our compiler and linker flags.

Sorry for conflicting requests and for requesting more changes, we are just trying to get this right.